core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -12,381 +12,397 @@ import com.unciv.models.UncivSound
 import com.unciv.models.UnitAction
 import com.unciv.models.UnitActionType
 import com.unciv.models.ruleset.Building
+import com.unciv.models.ruleset.unit.BaseUnit
 import com.unciv.models.translations.tr
 import com.unciv.ui.pickerscreens.ImprovementPickerScreen
 import com.unciv.ui.pickerscreens.PromotionPickerScreen
 import com.unciv.ui.utils.YesNoPopup
 import com.unciv.ui.worldscreen.WorldScreen
 
-class UnitActions {
+private const val CAN_BUILD_IMPROVEMENT = "Can build improvement: "
 
-    fun getUnitActions(unit: MapUnit, worldScreen: WorldScreen): List<UnitAction> {
+object UnitActions {
+
+    fun getUnitActions(unit: MapUnit, worldScreen: WorldScreen): Sequence<UnitAction> = sequence {
```

Comment:
here is the major change. Now it returns a sequence (since only place it's ever used is to map to buttons). It used to be used for searching as well, which sequence is good for, but like I said previously, this was replaced by grabbing the action directly instead of returning a list of UnitAction instances, searching the list for the right actions, then throwing away those instances and invoking the actions only.